src/freenas/usr/local/lib/middlewared_truenas/plugins/truenas.py

```diff
@@ -80,35 +82,36 @@ async def get_chassis_hardware(self):
         Really NFI about hardware at this point.  TrueNAS on a Dell?
         TRUENAS-UNKNOWN
         """
+        if TrueNASService.CHASSIS is not None:
```

Comment:
Can we just use `@functools.cache` instead of this?

Reply:
It doesn't work with our `@accepts()` decorator no matter how you stack them.

Reply:
It doesn't work with our `@accepts()` decorator no matter how you stack them.

Reply:
@yocalebo can you try using https://github.com/truenas/middleware/pull/8883 please?

Reply:
```
    @accepts()
    @returns(Str('system_chassis_hardware'))
    @cli_private
    @cache
    async def get_chassis_hardware(self):
```

Reply:
nice